scala2-functional-program-design/week2/streams/src/main/scala/streams/StringParserTerrain.scala

```diff
@@ -50,7 +50,9 @@ trait StringParserTerrain extends GameDef {
    * a valid position (not a '-' character) inside the terrain described
    * by `levelVector`.
    */
-  def terrainFunction(levelVector: Vector[Vector[Char]]): Pos => Boolean = ???
+  def terrainFunction(levelVector: Vector[Vector[Char]]): Pos => Boolean = (pos: Pos) =>
+    levelVector.lift(pos.row).getOrElse(Vector()).lift(pos.col).getOrElse('-') != '-'
```

Comment:
Not quite what I meant. I thought something like 
```
(pos.row < levelVector.length && pos.row >= 0 &&
        pos.col < levelVector.head.length && pos.col >= 0 &&
        levelVector(pos.row)(pos.col) != '-')
```

Reply:
Got it